.github/workflows/build-golang.yml

```diff
@@ -14,39 +14,51 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows], [macos-latest, macos] ]
+        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows-gnu], [macos-latest, macos] ]
     steps:
       - uses: actions/checkout@v2
       - name: Download workflow artifact
-        uses: dawidd6/action-download-artifact@v2.14.0
+        uses: dawidd6/action-download-artifact@v2.17.0
         with:
           workflow: "build-libs.yml"
           path: ./libs/${{ matrix.os-artifact[1] }}
           github_token: ${{ secrets.GITHUB_TOKEN }}
           name: ${{ matrix.os-artifact[1] }}
-      - name: Set up Go
+      - name: Download C header artifact
+        uses: dawidd6/action-download-artifact@v2.17.0
+        with:
+          workflow: "build-libs.yml"
+          path: ./libs/${{ matrix.os-artifact[1] }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          name: C_header
+      - name: Set up Go (default)
         uses: actions/setup-go@v2
         with:
           go-version: ^1.16
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: 3.9
-      - name: Build and Test Golang
+      # Set up Okapi-specific Go runtime. Includes setting GOLANG_LD_PATH env var.
+      - name: Set up Go (Okapi)
         run: |
           go version
           go install golang.org/x/lint/golint@latest
           go install github.com/jstemmer/go-junit-report@latest
           python ../../devops/build_sdks.py
+        shell: pwsh
+        working-directory: go/okapi
+      - name: Build and Test Golang
+        run: |
           go build
           golint
-          go test -v | go-junit-report > test_output.xml
+          go test -v -exec "env DYLD_LIBRARY_PATH=${{ env.GOLANG_LD_PATH }}" | go-junit-report > test_output.xml
         shell: pwsh
         working-directory: go/okapi
         env:
-          LD_LIBRARY_PATH: "${{ github.workspace }}/go/okapi"
-          DYLD_FALLBACK_LIBRARY_PATH: "${{ github.workspace }}/go/okapi"
-          DYLD_LIBRARY_PATH: "${{ github.workspace }}/go/okapi"
+          LD_LIBRARY_PATH: ${{ env.GOLANG_LD_PATH }}
```

Comment:
fix 3b.  we need to keep "the place we point LD to" and "the place where we copy lib artifacts to" in sync.   let's do this with an env var, which we can set in build_sdks.py and get in the GitHub workflow.

before this they were out of sync, e.g. on Windows you'd see `LD_LIBRARY_PATH=D:\a\okapi\okapi/go/okapi` - wonky slashes!  versus, `to_copy` dir in the script used the correct `D:\a\okapi\okapi\go\okapi`

Reply:
Good points. Windows is a matter of fact doesn't care about forward versus backward slashes. Yeah awkward/is. But it's better to be consistent and easier for us to update in one place.